unitTests/ui/OmniboxTest.ts

```diff
@@ -345,6 +345,55 @@ export function OmniboxTest() {
         test.cmp.magicBox.clear();
         expect(test.cmp.queryController.executeQuery).not.toHaveBeenCalled();
       });
+
+      it('triggerQueryOnClear should be forced to false if the search interface is configured to not allowQueriesWithoutKeywords', () => {
+        const advancedSetup = new Mock.AdvancedComponentSetupOptions(
+          null,
+          {
+            triggerQueryOnClear: true
+          },
+          env => {
+            env.searchInterface.options.allowQueriesWithoutKeywords = false;
+            return env;
+          }
+        );
+
+        test = Mock.advancedComponentSetup<Omnibox>(Omnibox, advancedSetup);
+        expect(test.cmp.options.triggerQueryOnClear).toBe(false);
+      });
+
+      it('triggerQueryOnClear should be forced to false if the search interface is configured to not allowQueriesWithoutKeywords', () => {
+        const advancedSetup = new Mock.AdvancedComponentSetupOptions(
+          null,
+          {
+            triggerQueryOnClear: true
+          },
+          env => {
+            env.searchInterface.options.allowQueriesWithoutKeywords = false;
+            return env;
+          }
+        );
+
+        test = Mock.advancedComponentSetup<Omnibox>(Omnibox, advancedSetup);
+        expect(test.cmp.options.triggerQueryOnClear).toBe(false);
+      });
+
+      it('triggerQueryOnClear should be forced to true if configured with search as you type', () => {
+        const advancedSetup = new Mock.AdvancedComponentSetupOptions(
+          null,
+          {
+            triggerQueryOnClear: false,
+            enableSearchAsYouType: true
+          },
+          env => {
+            env.searchInterface.options.allowQueriesWithoutKeywords = true;
```

Comment:
Hmmm, is it necessary for `allowQueriesWithoutKeywords` to be `true`? If so, then I think you should mention it in the test description.